File: challenges/2020/05-binaryBoarding/python/partOne.py
from common import *


def partOne(instr: str) -> int:
    input_list = parse(instr)

    highest_seat_id = 0
    for seat in input_list:
        parsed_seat = parse_seat(seat)
        seat_id = get_seat_id(*parsed_seat)
        if seat_id > highest_seat_id:
            highest_seat_id = seat_id

    return highest_seat_id

File: challenges/2020/05-binaryBoarding/go/challenge/partOne.go
package challenge

func PartOne(instr string) int {
	inputSlice := parse(instr)

	var highestSeatId int
	for _, seat := range inputSlice {
		psr, psc := parseSeat(seat)
		seatId := getSeatId(psr, psc)
		if seatId > highestSeatId {
			highestSeatId = seatId
		}
	}

	return highestSeatId
}
